### PR TITLE
appIndicator: Refactor named icons lookup and loading

### PR DESCRIPTION
**Pull request by 3v1n0**
From main commit:

```
We used to always looking up for the icons ourself, somewhat duplicating
the shell work, it allowed to prevent shell caching of some files, but still
we can remove lots of deprecated code and handle things more smartly given
that texture-caching can be good in case of well-known files.

In short:
 - named-only icons are now loaded through a Gio.ThemedIcon (and so cached)
 - icons with custom theme are searched in provided path and behave just
   like any other icon with full path as explained below.
 - icons with a full icon path (provided by the theme or not) may be loaded
   in different ways:
   * Using a Gio.FileIcon if they are in non-writable areas (so we assume
     that there are only a limited number of them) and they are cashed.
   * using a StTextureCacheSkippingFileIcon in newer gjs versions and so
     basically they're non-cachable Gio.FileIcon's
   * using a GdkPixbuf, potentially loaded via the icon theme (for svgs) in
     older gjs versions, and so that's not cached.
```

### CODE REVIEW

**Comment by 3v1n0** [comment]
@patricklibert @JOH451 @stuarthayhurst @luca-e075e @yochananmarqos

Since you've recently experienced #410 could you please if this keeps all working properly to you?
(See https://github.com/ubuntu/gnome-shell-extension-appindicator/#installation)

Thanks!

**Comment by stuarthayhurst** [comment]
Works perfectly for me

**Comment by yochananmarqos** [comment]
Yes, it's working. :+1:

**Comment by patricklibert** [comment]
Seems to be working...

**Comment by 3v1n0** [comment]
@luca-e075e it seems a different issue then given that's not related to the icon itself, but to the menu... That could be due to #416 maybe?

**Comment by 3v1n0** [comment]
@luca-e075e please open a new bug about it and provide logs if you get errors.

I personally get the vorta icon working properly.
